src/bt_peer_connection.cpp

```diff
@@ -366,7 +366,7 @@ namespace libtorrent {
 #ifndef TORRENT_DISABLE_LOGGING
 		peer_log(peer_log_alert::outgoing_message, "HAVE_ALL");
 #endif
-		char msg[] = {0,0,0,1, msg_have_all};
+		const char msg[] = {0,0,0,1, msg_have_all};
```

Comment:
for all the arrays used for messages, would you mind making them: ``static char const msg[]``?